specs/deneb/validator.md

```diff
@@ -133,47 +133,44 @@ use the `payload_id` to retrieve `blobs`, `blob_kzg_commitments`, and `blob_kzg_
 via `get_payload(payload_id).blobs_bundle`.
 2. Set `block.body.blob_kzg_commitments = blob_kzg_commitments`.
 
-#### Constructing the `SignedBlobSidecar`s
+#### Constructing the `BlobSidecar`s
 
 *[New in Deneb:EIP4844]*
 
-To construct a `SignedBlobSidecar`, a `signed_blob_sidecar` is defined with the necessary context for block and sidecar proposal.
+To construct a `BlobSidecar`, a `blob_sidecar` is defined with the necessary context for block and sidecar proposal.
 
 ##### Sidecar
 
-Blobs associated with a block are packaged into sidecar objects for distribution to the network.
+Blobs associated with a block are packaged into sidecar objects for distribution to the associated sidecar topic, the `blob_sidecar_{subnet_id}` pubsub topic.
 
 Each `sidecar` is obtained from:
 ```python
-def get_blob_sidecars(block: BeaconBlock,
+def get_blob_sidecars(signed_block: SignedBeaconBlock,
                       blobs: Sequence[Blob],
                       blob_kzg_proofs: Sequence[KZGProof]) -> Sequence[BlobSidecar]:
+    block = signed_block.message
+    block_header = BeaconBlockHeader(
+        slot=block.slot,
+        proposer_index=block.proposer_index,
+        parent_root=block.parent_root,
+        state_root=block.state_root,
+        body_root=hash_tree_root(block.body),
+    )
+    signed_block_header = SignedBeaconBlockHeader(message=block_header, signature=signed_block.signature)
     return [
         BlobSidecar(
-            block_root=hash_tree_root(block),
             index=index,
-            slot=block.slot,
-            block_parent_root=block.parent_root,
             blob=blob,
             kzg_commitment=block.body.blob_kzg_commitments[index],
             kzg_proof=blob_kzg_proofs[index],
+            commitment_inclusion_proof=compute_commitment_inclusion_proof(
```

Comment:
In the light client spec, we defined a more general `compute_merkle_proof_for_state(state: BeaconState, index: GeneralizedIndex)` helper.

It will be more compatible if we rename it to `compute_merkle_proof_for_beacon_block_body(body: BeaconBlockBody, index: GeneralizedIndex) -> Sequence[Bytes32]`.

Reply:
`compute_merkle_proof` sounds best to me.

Reply:
> It will be more compatible if we rename it to `compute_merkle_proof_for_beacon_block_body(body: BeaconBlockBody, index: GeneralizedIndex) -> Sequence[Bytes32]`.

`compute_merkle_proof_for_block_body` sounds more compatible, right? because it's not `compute_merkle_proof_for_beacon_state` in the light client spec.

Reply:
implemented in https://github.com/ethereum/consensus-specs/pull/3531/commits/545066c37572b1cac20f376ac3fd26a6e16e16ec